submissions/maryOkosun/phpSlim/app/controllers/ArticleController.php

```diff
@@ -0,0 +1,182 @@
+<?php
+
+namespace App\Controller;
+
+use DB\DB;
+use PDO;
+use PDOException;
+use Psr\Http\Message\ResponseInterface as Response;
+use Psr\Http\Message\ServerRequestInterface as Request;
+
+class ArticleController
+{
+    private $db;
+
+    public function __construct()
+    {
+        $this->db = (new DB())->connect();
+    }
+
+    /*public function index(Request $request, Response $response, $args)
+    {
+        try {
+            $query = $this->db->prepare("select *
+            from articles
+            inner join admin
+            on articles.id = admin.id
+            where articles.created_by = admin.name;");
+            $query->execute();
+            $article = $query->fetchAll();
+            $response->getBody()->write(json_encode($article));
+            return $response->withHeader('Content-Type', 'application/json')->withStatus(200);
+        } catch (PDOException $ex) {
+            $response->getBody()->write(json_encode(["error" => $ex->getMessage()]));
+            return $response->withHeader('Content-Type', 'application/json')->withStatus(500);
+        }
+    }*/
+
+    public function getArticle(Request $request, Response $response, $args)
+    {
+        $id = $args['id'];
+        try {
+            $query = $this->db->prepare("select *
+            from articles
+            inner join admin
+            on articles.id = admin.id
+            where articles.id = :id;");
+            $query->bindParam(":id", $id);
+            $query->execute();
+            $article = $query->fetch();
+
+            if (!$article) {
+                $response->getBody()->write(json_encode(["error" => "Article not found"]));
+                return $response->withHeader('Content-Type', 'application/json')->withStatus(404);
+            }
+
+            $response->getBody()->write(json_encode($article));
+            return $response->withHeader('Content-Type', 'application/json')->withStatus(200);
+        } catch (PDOException $ex) {
+            $response->getBody()->write(json_encode(["error" => $ex->getMessage()]));
+            return $response->withHeader('Content-Type', 'application/json')->withStatus(500);
+        }
+    }
+
+    public function create(Request $request, Response $response, $args)
+    {
+        $requestData = $request->getParsedBody();
+        $title = $requestData['title'];
+        $description = $requestData['description'];
+        $created_by = $requestData['created_by'];
+        $createdAt = date('Y-m-d H:i:s');
+
+        $sql = "INSERT INTO articles (title, description, status, created_by, created_at) VALUES (:title, :description, :status, :created_by, :created_at)";
+
+        try {
+            $query = $this->db->prepare($sql);
+            $query->bindParam(':title', $title);
+            $query->bindParam(':description', $description);
+            $query->bindParam(':status', $status);
```

Comment:
This will even throw an error. Always try running what you did again if you make corrections. If you do, you would have noticed this error. You won't have to go back and forth

Reply:
I actually ran the code but the status was showing null. I did not notice but currently it is reflecting the  default UNPUBLISHED.